app/controllers/TaskListController.scala

```diff
@@ -60,8 +63,39 @@ class TaskListController @Inject() (
 
   def onSubmit(lrn: LocalReferenceNumber): Action[AnyContent] = actions
     .requireData(lrn)
-    .andThen(checkDependentTaskCompleted[PreTaskListDomain]) {
-      _ => ???
+    .andThen(checkDependentTaskCompleted[PreTaskListDomain])
+    .async {
+      implicit request =>
+        // Either[ReaderError, A], TODO - move to service layer
+
+        val getData: Future[domain.EitherType[DepartureDomain]] = for {
+          ctcCountries                          <- countriesService.getCountryCodesCTC()
+          customsSecurityAgreementAreaCountries <- countriesService.getCustomsSecurityAgreementAreaCountries()
+        } yield DepartureDomain
+          .userAnswersReader(ctcCountries.countryCodes, customsSecurityAgreementAreaCountries.countryCodes)
+          .run(request.userAnswers)
+
+        // TODO - another service layer
+        val submit = getData.flatMap(
+          _.traverse(apiService.submitDeclaration(_))
+        )
+
+        submit.flatMap {
+          case Right(value) =>
+            value.status match {
+              case status if is2xx(status) =>
+                Future.successful(Redirect(controllers.routes.DeclarationSubmittedController.onPageLoad()))
+              case status if is4xx(status) =>
+                // TODO - log and audit fail. How to handle this?
+                Future.successful(BadRequest)
+              case _ =>
+                // TODO - log and audit fail. How to handle this?
+                Future.successful(InternalServerError("Something went wrong"))
+            }
+          case Left(e) =>
+            // TODO - log and audit fail. How to handle this?
+            Future.successful(InternalServerError("Something went wrong"))
+        }
```

Comment:
Probably not needed for this PR, particularly if we're looking to split some of this out into different service layers, but there's cool stuff you can do with the cats library for code like this. Something like:

```
(for {
          ctcCountries <- EitherT.right(countriesService.getCountryCodesCTC())
          customsSecurityAgreementAreaCountries <- EitherT.right(countriesService.getCustomsSecurityAgreementAreaCountries())
          data = DepartureDomain.userAnswersReader(ctcCountries.countryCodes, customsSecurityAgreementAreaCountries.countryCodes).run(request.userAnswers)
          response <- EitherT(data.traverse(apiService.submitDeclaration(_)))
        } yield {
          response.status match {
            case status if is2xx(status) =>
              Redirect(controllers.routes.DeclarationSubmittedController.onPageLoad())
            case status if is4xx(status) =>
              // TODO - log and audit fail. How to handle this?
              BadRequest
            case _ =>
              // TODO - log and audit fail. How to handle this?
              InternalServerError("Something went wrong")
          }
        }).value.map {
          case Right(value) =>
            value
          case Left(e) =>
            // TODO - log and audit fail. How to handle this?
            InternalServerError("Something went wrong")
        }
```